src/odemis/driver/xt_client.py

```diff
@@ -2179,16 +2205,42 @@ def __init__(self, name: str, role: str, parent: model.HwComponent, rng: dict =
                                                 readonly=True)
         self._updatePosition()
 
+        # update the offset if raw stage coordinates are read
+        self._get_offset_raw_coordinate_system()
+
         # Refresh regularly the position
         self._pos_poll = util.RepeatingTimer(5, self._refreshPosition, "Stage position polling")
         self._pos_poll.start()
 
+    def _get_offset_raw_coordinate_system(self) -> None:
+        """Calculate the offset in linear stage axes x and y when the stage coordinates are read in raw coordinates such
+         that it is equal to x and y when stage coordinates are read in linked coordinates."""
+        if self.raw:
+            self.parent.set_raw_coordinate_system(self.raw)
+            pos = self._getPosition()
+            logging.debug(f"Raw stage coordinates {pos}")
+            self.parent.set_raw_coordinate_system(False)
+            pos_after = self._getPosition()
+            logging.debug(f"Linked stage coordinates {pos_after}")
+            for axis in self.offset.keys():
+                self.offset[axis] = pos_after[axis] - pos[axis]
+            # the offset should only be in linear axes, it is not expected in rotational axes
+            assert all(pos[axis] == pos_after[axis] for axis in ["rx", "rz"])
+            logging.debug(f"The offset values in x and y are {self.offset} when stage is in the raw coordinate system")
```

Comment:
Please only put 1 logging message for this function. It should say the non-raw & raw values, and conclude on the offset, as a single sentence.